### ISSUE TITLE
Routine for confirming the existence of a building, or marking it as demolished

### ISSUE THREAD

**Issue by astaninv** [Zverik/every_door#691]
Currently, there is an excellent procedure that allows to mark POIs as checked, closed, or non-existent. However, there is no similar functionality to check whether buildings still exist or are already demolished. 

In earlier versions of the program, it was possible to delete the entire building, however, this approach is incorrect, since it leaves garbage in the database and does not tell other cartographers about the reason for the absence of the building, which can still be visible on satellite images.

I suggest the following procedure. Two buttons on the building editing screen, with functions:
1) Green, by which the **[check_date=*](https://wiki.openstreetmap.org/wiki/Key:check_date)** tag with the current date will be assigned or updated to the building's tag list, thus, confirming  existence of the building.
2) Yellow, change of building status. Three main statuses should be implemented:
2.1) Disused, but still intact building is marked as **[disused=yes](https://wiki.openstreetmap.org/wiki/Key:disused:*)**
2.2) Partially collapsed, but with intact walls or other easy visible remains: **[ruins=yes](https://wiki.openstreetmap.org/wiki/Key:ruins)**
2.3) A demolished building, from which only the little traces remains - main building tag substituted with **[demolished:building=*](https://wiki.openstreetmap.org/wiki/Key:demolished:building)**

The advantages of this approach are that no changes are made to the contours, only to the tags; physically existing buildings are displayed correctly in all renderings; cartographers see that the building existed, but was demolished. This option would be useful for a cities that are actively being rebuilt, where the demolition of old buildings and the construction of new buildings occur regularly. Functions  2.1 and 2.2 are optional, and could be omitted.

**Comment by mnalis** [comment]
> Green, by which the [check_date=*](https://wiki.openstreetmap.org/wiki/Key:check_date) tag with the current date will be assigned or updated to the building's tag list, thus, confirming existence of the building.

that might be problematic, especially as building often have other things tagged on the same polygon (e.g. `building=yes`+`shop=supermarket`+`name=*`+`opening_hours=*`+....). Adding `check_date` would mean all of the tags were checked, not just building existence. 

So, if building is still there, you'd need to verify also if supermarket is there, if that is the same supermarket with same name, if its opening hours are the same etc. before adding `check_date=*` on it. 

One might work around that by using `check_date:building=*` instead, but what would be the exact purpose of adding that? In amenities mode, EveryDoors uses `check_date`  to show green checkmark if it has been checked in the last two weeks so user could save time by skipping it without rechecking, but that does not seem to be as useful for building mode (not only are buildings usually demolished much less often then every two weeks, but also EveryDoor buildings mode does not show such checkmarks anyway).

----

> Currently, there is an excellent procedure that allows to mark POIs as checked, closed, or non-existent. However, there is no similar functionality to check whether buildings still exist or are already demolished.

This I'd find useful too! One currently has to resort to raw tag editor in EveryDoor, which is quite cumbersome. It would be nice to be able to change that `building=*` to `abandoned:building=*` or `demolished:building=*` more easily. 

However there are also problems with that: one is mentioned above, e.g. if building polygon has other tags like `shop=*` tagged on same element, but also if there were multiple nodes and/or smaller polygons _inside_ that demolished building -- they would in most cases need to be cleaned up too when building is demolished, and would be too easy to miss -- unless there was some EveryDoor AI which would detect all nodes and sub-polygons inside current polygon and handled them correctly automagically (after confirming with user?). So (to me) it seems to become rather complex and error prone.

So, on second thought, **perhaps best solution when building is disused/abandoned/demolished is just to leave a note** (detailing if any of the POIs inside / on the walls survived, like e.g. postboxes or ATMs or trash cans) and then postprocess that building later in full-fledged [editor](https://wiki.openstreetmap.org/wiki/Editors) (like JOSM/iD/Vespucci). Hopefully they don't get demolished all that often for such workflow to become too burdensome.

**Comment by pandadeepimpact** [comment]
> So, on second thought, perhaps best solution when building is disused/abandoned/demolished is just to leave a note (detailing if any of the POIs inside / on the walls survived, like e.g. postboxes or ATMs or trash cans) and then postprocess that building later in full-fledged [editor](https://wiki.openstreetmap.org/wiki/Editors) (like JOSM/iD/Vespucci).

But this would require users of being aware of such tags in the first place. I've actually come across some ruins once, but never thought that there's a tag specifically for it.

How about adding a "mark defunct" button on the buildings layer, but instead of immediately adding the `disused` tag, it would lead to a note with those choices (`disused`, `ruins`, `demolished`)?

**Comment by astaninv** [comment]
> How about adding a "mark defunct" button on the buildings layer, but instead of immediately adding the `disused` tag, it would lead to a note with those choices (`disused`, `ruins`, `demolished`)?

Agree, this might be a good solution, at least to start with. The only disadvantage of leaving a note instead of directly changing building=* tag is that you still have to find some more free time to run JOSM and fix tags once again.

**Comment by Zverik** [comment]
In 5.1 you can mark a building as demolished with a button on its editor page. That is, tap on a building, tap "More...", and then tap "Demolished". For ruins, choose "ruins" from the building type combo.
